src/lib.rs

```diff
@@ -1215,4 +1235,24 @@ mod tests {
                      "  bar"];
         assert_eq!(dedent(&add_nl(&x)), add_nl(&y));
     }
+
+    #[test]
+    fn multiline() {
+        assert_eq!(fill("1 3 5 7\n1 3 5 7",11),"1 3 5 7\n1 3 5 7");
+        assert_eq!(fill("1 3 5 7\n1 3 5 7",5),"1 3 5\n7\n1 3 5\n7");
+        assert_eq!(fill("1 3 5 7\n1 3 5 7",6),"1 3 5\n7\n1 3 5\n7");
```

Comment:
It's a little hard for me to figure out which corner-case each test is testing :-) Could you perhaps add multiple smaller tests if there is a common theme?

Here, I think a third line with
```
        assert_eq!(fill("1 3 5 7\n1 3 5 7", 7), "1 3 5 7\n1 3 5 7"); 
```
could be interesting -- it would demonstrate that the code doesn't accidentally add an extra `\n`.